### ISSUE TITLE
Mutator dialog UI with empty sub-block list causes issues

### ISSUE THREAD

**Issue by Marclev78** [google/blockly#1759]
### Mutator dialog with empty flyout  causes exception and rendering issues

I am trying to define a mutator dialog that only has a single top level block, for example if I can configure the block using check boxes on that top level block, so don't need any others for the dialog.

In JavaScript, I try to initialise such a mutator like this:
```
Blockly.Extensions.registerMutator("precondition_configuration", {
  mutationToDom: function () {
    ...
  },

  domToMutation: function (xmlElement) {
    ...
  },
  compose: function (topBlock) {
    ...
  },

  decompose: function (workspace) {
    const topBlock = workspace.newBlock("mutator-config-block");

    topBlock.initSvg();
    topBlock.render();

    return topBlock;
  }
}, 
() => {
}, []); // <== Empty list of blocks for mutator toolbox
```

I would expect the resulting mutator dialog to just show me the top block "`mutator-config-block` without any flyout (or an empty toolbox, if necessary).

### Actual Behavior

Opening the mutator dialog causes this exception:

```
blockly_compressed.js:1227 Uncaught TypeError: Cannot read property 'contentHeight' of null
    at Blockly.Mutator.resizeBubble_ (blockly_compressed.js:1227)
    at Blockly.Mutator.setVisible (blockly_compressed.js:1230)
    at Blockly.Mutator.Blockly.Icon.iconClick_ (blockly_compressed.js:948)
    at Blockly.Mutator.iconClick_ (blockly_compressed.js:1223)
    at SVGGElement.h (blockly_compressed.js:1631)
```

A mutator dialog does then appear on screen, however it extends to the edges of the workspace (although it does contain the single top block as expected).

I have to put an artificial block into the mutator toolbox to work around this, which is both ugly, and causes user confusion as they don't know what to do with the artificial block.

**Comment by AnmAtAnm** [comment]
@Marclev78 The toolbox area is also the area used to delete mutator blocks. Are you intending to introduce other ways of creating blocks that may need to be deleted?

**Comment by rachel-fenichel** [comment]
This is a valid use case but high effort for us to fix.  We will keep the issue open so that we can find out if more people are running into it.

**Comment by nederland074** [comment]
@Marclev78 Hi, I've also struggled with this problem recently and found a workaround, which is to translate this artificial block beyond the borders of the mutator dialog window, something like this:
```
Blockly.Blocks['dummyBlock'] = {
    init: function() {
        this.translate(-9999, -9999);
    }
}
```

**Comment by picklesrus** [comment]
Another vote for fixing this came up on the discussion list: https://groups.google.com/forum/#!topic/blockly/qSFI9scMMRQ . With 3 reports in 3 months, I'm going to take off the low priority tag?

Red Panda found a clever way, though still non-ideal way that is maybe easier than the translate(-9999, -99999). They sent  block list: [''].  You still get a flyout though :-/

**Comment by Marclev78** [comment]
> @Marclev78 The toolbox area is also the area used to delete mutator blocks. Are you intending to introduce other ways of creating blocks that may need to be deleted?

Hi, just realized I never responded to this, sorry! So, in the original use case I simply had a block that has a list of checkbox inputs in the mutator UI, so there's no need to drag anything into the block being configured.

The state of the checkbox inputs is read in the mutation methods to either add or remove sections on the original block dynamically.

**Comment by BeksOmega** [comment]
Fixed by #3958
